### ISSUE TITLE
Change font in Start Menu

### ISSUE THREAD

**Issue by SkyTrax79** [ramensoftware/windows-11-start-menu-styling-guide#46]
Is it possible to change the font of the items of the Start Menu like we can do with the Taskbar ?  It's the first time I find a mod that allows to change the taskbar font easily, that would be great if we can also change it in the Start Menu and the context menu of the taskbar.

Thanks

**Comment by bbmaster123** [comment]
yes it is possible, and it works the same way. find the element/elements you want to style, for example the target TextBlock[2] will apply to most of the start menu text, and simply set in the style box FontFamily=fontname

As for context menu, you can only adjust elements that are styled with xaml, so jumplists, toast notifications, and the small context menu with taskbar settings and task manager are all on the table by also using the notification styler mod, but none of the win32 context menus.

Hope that helps, please reach out if you need more help :)

**Comment by SkyTrax79** [comment]
Thank you so much, I've been looking for a long time to see if there's a way to change the font in these places.

This is all because I changed the fonts with the Winaero tweaker application, there are choices for Icons, Menus, Messages fonts, Status bar font, Windows title bar and System font which only change in a few places.

I've noticed that Menu Fonts only changes the old-style menus font, which I assume to be Win32.

**Comment by SkyTrax79** [comment]
Ok so what would be the target to change the font of the Settings app ?  It's by far the "modern" app that I use and see the most.  And do you have to reboot or restart Explorer for the changes to take effect ?  Sorry for the noob questions but in the case of the taskbar labels the first time I tried it changed only the Explorer window tab label, others changed later.

**Comment by SkyTrax79** [comment]
What I understand from experimenting is that you can't enter Targets intended for the Start Menu, for example, in the mod settings for the Tassbar and vice-versa?

I've tried UWPspy for the Settings application and I can see where the settings are for example the word “Settings” on top or the words “Find a setting” in the search box.  In which mod should I enter these targets?  That's why I was asking if you can enter new targets in any mod.

**Comment by bbmaster123** [comment]
good questions

1. yes thats right, WinaeroTweaker changes the fonts used in your msstyles theme, which is to say that it changes win32 fonts. Start menu, taskbar, Settings, etc are not win32 so winaerotweaker doesn't affect these

2. Settings requires the mod to actually target the settings app's process to inject into, which you can do either by adding the exe to the include list in advanced, or by forking the mod in Windhawk, and changing the included process to the desired exe. for settings, that is SystemSettings.exe. This can also be almost any other uwp app as well, see here for some examples https://github.com/bbmaster123/Themes-Showcase/tree/main/Damascus%20Show%20Case

3.  enter the targets into the mod with the include in advanced, or in your forked version of the mod

**Comment by SkyTrax79** [comment]
Thanks for the answer, it's interesting to understand how it works.

So far I've been able to change the font of the taskar Context Menus, Notification Panel and Start Menu except where it says “Pinned” “All apps” and “Back”, I need to figure out how to change the kind of tooltip above the taskabr thumbnails too.

I searched for the Settings app and we can see, as I said at the beginning, where it says “Settings” at the top in Windows.UI.Xaml.Controls. TextBlock - SettingsLabel, on the right in Attributes it says “Text:  Settings” without the FontFamily entry.  I'd have to find a way to make it say that I want all ...Xaml.Controls.TextBlock with this Fontfamily, I don't know if it's something that can be added, I'm just experimenting to find out how it works.

**Comment by bbmaster123** [comment]
absolutely, I agree

it can definitely be tricky sometimes. The style you want isn't always available, it might be in another target or it just might not be accessible by the mod, sometimes you get lucky and sometimes not. 

have you tried setting it in windhawk anyways? sometimes windhawk can set something that uwpspy has trouble with, in my experience. I tried for TextBlock#PinnedListHeaderText and it works. 

The target for Tooltip is ` Tooltip` and setting fontfamily on that also works

**Comment by SkyTrax79** [comment]
Ah thanks for tooltip :)  Yeah it can be tricky,  I checked all the SystemSettings.exe and there is no place where it says FontFamily, at other places I found FontFamily (null) and it does nothing if you put something there, also when you change the fonts of the notification panel it breaks the small icon beside 30 min. at the bottom - + and Focus because they share the same name, not a big deal.  

I wonder how you found that writing "TextBlock[2]" would change all the fonts of the Start Menu ?  I've tried it at other places and it does nothing, but it's fun because you can try anything and nothing breaks lol, the worst that could happens, at least in my experiments, is that it does nothing and you just have to remove the bad target.

**Comment by SkyTrax79** [comment]
Like for the tooltip should I enter TextBlock#Tooltip ? or Tooltip[2] ?

I've tried to make a fork of the Taskbar Styler mod, since I know almost nothing about this coding and didn't have the time to read all the code lines when I've done it, I only change the explorer.exe at the top for SystemSettings.exe, I've compiled it and the mod worked but that's it.

**Comment by bbmaster123** [comment]
hey,
sorry for the delay, I guess I got busy

a number in square brackets like [2] tells the mod to apply the style to the #child of the written target, so in this case it is targeting the 2nd child down the visual tree under rectangle, whatever it happens to be named.

As for which you should use, either should work equally well, but if for some reason a target is giving trouble or you are looking for a particular target, using the number can help 

the method you used to fork your mod is correct, you just change explorer.exe to whichever exe(s) you need and recompile. nothing else to worry about after that except styling! :)

cheers!

**Comment by SkyTrax79** [comment]
Ah ok, thanks a lot, I understand now 😀.

Obtenir Outlook pour Android<https://aka.ms/AAb9ysg>
________________________________
